shared/js/template.js

```diff
@@ -113,25 +113,43 @@
    *                          - safe, a list of properties that contain
    *                          HTML that is known and are
    *                          "known" to ignore.
-   */
-  Template.prototype.interpolate = function(data, options) {
-    // This _should_ be rewritten to use Firefox's support for ES6
-    // default parameters:
-    // ... = function(data, options = { safe: [] }) {
-    //
-    options = options || {};
-    options.safe = options.safe || [];
+  */
+  Template.prototype.interpolate = function(data = {}, options = { safe: [] }) {
+    var safeList = options.safe;
+    var hasSafeList = Boolean(safeList);
+
+    if (hasSafeList && !Array.isArray(safeList)) {
+      throw new TypeError(
+        `"options.safe" must be an Array, not ${typeof options.safe}`
+      );
+    }
+
+    var hasSafeProperties = hasSafeList && safeList.length;
 
     return this.extract().replace(rmatcher, function(match, property) {
       property = property.trim();
-      // options.safe is an array of properties that can be ignored
-      // by the "suspicious" html strategy.
-      return options.safe.indexOf(property) === -1 ?
-        // Any field that is not explicitly listed as "safe" is
-        // to be treated as suspicious
-        Template.escape(data[property]) :
-        // Otherwise, return the string of rendered markup
-        data[property];
+
+      var isPropertySafe = hasSafeProperties && safeList.includes(property);
+
+      if (!(property in data)) {
+        throw new
+          Error(`No value provided for template identifier "${property}"'`);
```

Comment:
nit: I think there is a `'` that should not be here.

also, if you need to go to the next line, please do something like:
```
throw new Error(
  '.....'
);
```